jina/excepts.py

```diff
@@ -71,3 +72,37 @@ class NoContainerizedError(Exception, BaseJinaException):
 
 class PortAlreadyUsed(RuntimeError, BaseJinaException):
     """Raised when to use a port which is already used"""
+
+
+class NetworkError(Exception, BaseJinaException):
+    """Raised when communication between microservices fails"""
+
+    def __init__(
+        self,
+        og_exception: grpc.aio.AioRpcError,
+        request_id: str = '',
+        dest_addr: str = '',
+        details: str = '',
+    ):
+        """
+        :param og_exception: the original exception that caused the network error
+        :param request_id: id of the request that caused the error
+        :param dest_addr: destination (microservice) address of the problematic network call
+        :param details: details of the error
+        """
+        self.og_exception = og_exception
+        self.request_id = request_id
+        self.dest_addr = dest_addr
+        self._details = details
+
+    def code(self):
+        """
+        :return: error code of this exception
+        """
+        return self.og_exception.code()
+
+    def details(self):
```

Comment:
Consciously not making these `@property`, this way this error shares some of its interface with `AioRpcError`. That makes downstream handling easier.

Reply:
I do not like this design, this is by chance that it has the same method or attribute, but if this is what u expect, u should use some inheritance, otherwise it is too risky to have this.